### PR TITLE
add troll tests and troll functionality

### PR DESCRIPTION
**Pull request by brandonhutchins**

### CODE REVIEW

**Review by jcpirtle — APPROVED** [review]

File: source/scripts/data/monsters.js [hunk 1]
```diff
@@ -36,6 +36,13 @@ export default {
         color: DATA.COLORS.WHITE,
         health: 5,
         strength: 5,
+        turnCounter: function() {
+            this.somecount = this.somecount + 1 || 0
```
Comment (ehgoodenough) [inline source/scripts/data/monsters.js]: Is `this.somecount` the best variable name for this?

File: source/scripts/data/monsters.js [hunk 2]
```diff
@@ -36,6 +36,13 @@ export default {
...
+            if(this.somecount % 4 == 0) {
+                return this.phase = true
+            }
+            return this.phase = false
```
Comment (ehgoodenough) [inline source/scripts/data/monsters.js]: It'll be easier (in my opinion) to read if this `if` statement just sets `this.phase`, and then the function has a single `return this.phase` at the bottom.

File: source/scripts/tests/TrollTest.js [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+            {
+                position: {x: 1, y: 1},
+                protomonster: MONSTERS.WHITE_TROLL
+            }
+        ]
+
+        // Also, by purposefully forgetting to
+        // pass a "wave" to the game, our test
+        // won't be interrupted by other monsters
+        // being spawned in.
```
Comment (ehgoodenough) [inline source/scripts/tests/TrollTest.js]: I put these comments in here for your edification; please delete them after you're done.

File: source/scripts/tests/TrollTest.js [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+    expect(game.monsters[0].position).to.eql({x: 1, y: 2})
+    game.monsters[0].onAction()
+    expect(game.monsters[0].position).to.eql({x: 1, y: 2})
+    game.monsters[0].onAction()
+    
+    Expect(game.monsters[0].position).not.to.eql({x: 1, y: 1})
+    Expect(game.monsters[0].position.x).not.to.be.below(0)
+    Expect(game.monsters[0].position.y).not.to.be.below(0)
+    Expect(game.monsters[0].position.x).not.to.be.above(DATA.FRAME.WIDTH)
+    Expect(game.monsters[0].position.y).not.to.be.above(DATA.FRAME.HEIGHT)
```
Comment (ehgoodenough) [inline source/scripts/tests/TrollTest.js:45]: The tests look good. We really need to add a test case for attacking the adventurer, but of course, we don't even have adventurer health yet, so it's not like we can do that easily.

**Review by ehgoodenough — COMMENTED** [review]
Looking good! A couple of outstanding issues, but I'd merge it otherwise.

Also, next time, can we get someone other than @jcpirtle and @brandonhutchins to keep reviewing each other's changes without comments?

**Comment by ehgoodenough** [comment]
Also, could you please provide a test link? We've been slacking off on providing those.

**Comment by brandonhutchins** [comment]
https://brandonhutchins.github.io/enchiridion/trolltest/

File: source/scripts/model/Monster.js [hunk 5]
```diff
@@ -116,7 +116,9 @@ export default class Monster {
         this.health -= damage
         if(this.health <= 0) {
             this.game.remove("monsters", this)
-
+            if(this.basesprite == DATA.SPRITES.MONSTERS.BLUE_SLIME) {
+                console.log("yay you killed a blue lime")
+            }
```
Comment (ehgoodenough) [inline source/scripts/model/Monster.js:121]: Was this meant to be left in?

**Comment by ehgoodenough** [comment]
The build looks great, and the code works fine! Ship it!

selfie-1

**Pull request merged**